vowpalwabbit/core/include/vw/core/example.h

```diff
@@ -35,6 +35,12 @@ namespace VW
 void copy_example_data(example* dst, const example* src);
 void setup_example(VW::workspace& all, example* ae);
 
+void add_example_namespace(VW::example& ec, VW::namespace_index ns, const features& fs);
```

Comment:
These functions are already a bit spooky because the del is actually just a truncate based on the size of the given features. Renaming to insert/erase seems reasonable though

Reply:
I think append and truncate better indicate what exactly they do, especially since you cannot give a position to insert at